uint64 from_id
uint64 to_id
---
float64 x
float64 y
float64 z